libs/graphslam/include/mrpt/graphslam/CGraphSlamEngine_impl.h

```diff
@@ -119,7 +110,7 @@ void CGraphSlamEngine<GRAPH_T>::getRobotEstimatedTrajectory(
 	typename GRAPH_T::global_poses_t* graph_poses) const
 {
 	MRPT_START;
-	ASSERT_(graph_poses);
+	ASSERTDEB_(graph_poses);
 	std::lock_guard<std::mutex> m_graph_lock(m_graph_section);
 	*graph_poses = m_graph.nodes;
```

Comment:
This is a copy right?
``` 
template <class GRAPH_T>
 void CGraphSlamEngine<GRAPH_T>::getRobotEstimatedTrajectory(
 	typename GRAPH_T::global_poses_t* graph_poses) const
```
This signature IMO is better and eliminates the need for the assert. It shouldn't incur runtime penalties. It might even be faster, but I haven't proven that.
```
template <class GRAPH_T>
typename GRAPH_T::global_poses_t void CGraphSlamEngine<GRAPH_T>::getRobotEstimatedTrajectory() const
```

Reply:
The asserts in all of these places are just redundant anyway..
They should have been passed by reference in the first place.. 

However, passing the vector to be modified by reference and returning it seems equivalent for me though.. what makes you think that the latter may be faster?

Reply:
https://godbolt.org/g/ZvZtdo
```
#include <iostream>
struct pose_t
{
    pose_t()
    {
        std::cout << x;
        std::cout << y;
        std::cout << z;
    }
    pose_t(int x, int y, int z) :
    x(x), y(y), z(z)
    {
        std::cout << x;
        std::cout << y;
        std::cout << z;
    }
    int x;
    int y;
    int z;
};

inline void getPoseCopy(pose_t &t)
{
    static pose_t pose{1,2,3};
    t = pose;
}

inline pose_t getPoseCopy()
{
    static pose_t pose{5,6,7};
    return pose;
}

void test1()
{
 pose_t pose(getPoseCopy());
 std::cout << pose.x;
 std::cout << pose.y;
 std::cout << pose.z;
}

void test2()
{
 pose_t pose;
 getPoseCopy(pose);
 std::cout << pose.x;
 std::cout << pose.y;
 std::cout << pose.z;
}
```
For one, you can avoid the default constructor.
Copy/move elision makes it fast.
http://en.cppreference.com/w/cpp/language/copy_elision